src/NuGet.Server.Core/Infrastructure/ServerPackageStore.cs

```diff
@@ -77,45 +78,57 @@ public void Remove(string id, SemanticVersion version, bool enableDelisting)
             }
         }
 
-        public HashSet<ServerPackage> GetAll(bool enableDelisting)
+        public async Task<HashSet<ServerPackage>> GetAllAsync(bool enableDelisting, CancellationToken token)
         {
             var allPackages = new ConcurrentBag<ServerPackage>();
-            
-            Parallel.ForEach(_repository.GetPackages(), package =>
-            {
-                ServerPackage serverPackage;
 
-                // Try to create the server package and ignore a bad package if it fails.
-                if (TryCreateServerPackage(package, enableDelisting, out serverPackage))
-                {
-                    allPackages.Add(serverPackage);
-                }
-            });
+            var tasks = _repository
+                .GetPackages()
+                .Select(package => TryAddServerPackageAsync(allPackages, package, enableDelisting, token))
+                .ToList();
+
+            await Task.WhenAll(tasks);
 
             // Only return unique packages.
             return new HashSet<ServerPackage>(allPackages, IdAndVersionEqualityComparer.Instance);
         }
 
-        private bool TryCreateServerPackage(IPackage package, bool enableDelisting, out ServerPackage serverPackage)
+        private async Task TryAddServerPackageAsync(
+            ConcurrentBag<ServerPackage> allPackages,
+            IPackage package,
+            bool enableDelisting,
+            CancellationToken token)
         {
-            try
+            // Immediately defer work to the background thread.
+            await Task.Yield();
+
+            // Try to create the server package and ignore a bad package if it fails.
+            var serverPackage = CreateServerPackage(package, enableDelisting, token);
+            if (serverPackage != null)
             {
-                serverPackage = CreateServerPackage(package, enableDelisting);
+                allPackages.Add(serverPackage);
+            }
+        }
 
-                return true;
+        private ServerPackage CreateServerPackage(
+            IPackage package,
+            bool enableDelisting,
+            CancellationToken token)
```

Comment:
Good point. I'll fix this in a subsequent PR (will have one soon for SemVer 2.0.0 which is in master but not dev).